oauth2/generic.go

```diff
@@ -211,7 +189,7 @@ func (g *Generic) getPrimaryEmail(client *http.Client) (string, error) {
 
 func (g *Generic) primaryEmail(emails []*UserEmail) (string, error) {
 	for _, m := range emails {
-		if m != nil && m.GetPrimary() && m.GetVerified() && m.Email != nil {
+		if m != nil && m.Primary != nil && m.Verified != nil && m.Email != nil {
```

Comment:
should this be `m != nil && m.Primary != nil && m.Primary && m.Verified != nil && m.Verified && m.Email != nil`? Seems like we are ignoring the value of those fields i.e. if Primary is false then it is != nil but it is still not primary.

Reply:
good question. maybe `.Primary` and `.Verified` only exist if `true`? i'm not sure without investigating further. this PR reverts the code to where it's been for the past year where users never reported any issues, and i'm wary of making further changes in this coming release since 1) we haven't had any other issues reported in the past year, 2) a user who was affected by what this reverts just confirmed that this fixes their problem, and 3) i'm worried about breaking a bunch of of other generic oauth users without exhaustive testing. so i suggest investigating this as a separate issue / PR.

Reply:
👍

Reply:
Opened #3390.